### PR TITLE
PR for Lesson 7 HTML Rendering

### PR DESCRIPTION
**Pull request by LouReis**
Please review the work for Lesson07:

run_html_render.py
test_html_render.py
html_render.py
sample_html.py
test_html_output1.py

There are 2 tests that are not working and I'm not sure why. I'm not sure how to complete the Part 5 items involving attributes, please provide an example.  I'm confused about what needs to be done, even after reviewing the tutorial.

Feedback: This doesn't seem like a useful exercise at this juncture, seems like an exercise of medium complexity involving OO, Classes, Subclasses would be much more appropriate and give the student the opportunity to master those topics without so much confusion, file parsing, etc. Progress in smaller iterations is more helpful then high complexity & confusion...

Thanks & Regards,
Lou

### CODE REVIEW

**Comment by LouReis** [comment]
Hello,
I also submitted the exercise Circle Class for Lesson08.  Please review & grade.  The files included in the git push are:

test_circle_class.py
circle_class.py

Thanks & Regards,
Lou

File: students/LouReis/Lesson09/mailroom5.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+def enter_new_donor(donor, donation):
+    donations.update({donor:[donation]})
+    #print_letter(donor, donation)
+
+# This Option generates a thank you letter for a new donation and prints to the screen.
+def thanks_letter():
+    print('\nYou Chose Option 2\n\n')
+    print('Create a Thank You Letter\n\n')
+    donor = 'L'
+    while donor == 'L':
```
Comment (charlesrobison) [inline students/LouReis/Lesson09/mailroom5.py]: Recommend using lower() here to catch case sensitive input.

File: students/LouReis/Lesson09/test_mailroom5.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+donation_report()
+thanks_letter()
+thanks_letter_all()
+print_letter(donor, amount)
+quit()
+
+"""
+
+# referencing all of the functions, names, variables, symbols to import
+from mailroom4 import main_menu, donation_report, thanks_letter
```
Comment (charlesrobison) [inline students/LouReis/Lesson09/test_mailroom5.py]: Looks like you forgot to update these to mailroom5

**Comment by LouReis** [comment]
Thanks...good to know that the circle class is good.  I'm just getting started on Lesson9 mailroom (not ready to grade or look at), I emailed a few questions on it...

 

Lou

 

Sent: Saturday, January 12, 2019 at 11:50 AM

**Pull request merged**